Makefile

```diff
@@ -2,15 +2,111 @@
+	cp proto/serving.proto client/src/featureform/proto/serving.proto
+
+	protoc --go_out=. --go_opt=paths=source_relative     --go-grpc_out=. --go-grpc_opt=paths=source_relative     ./proto/serving.proto
+	python -m grpc_tools.protoc -I ./client/src --python_out=./client/src --grpc_python_out=./client/src/ ./client/src/featureform/proto/serving.proto
+
+	protoc --go_out=. --go_opt=paths=source_relative     --go-grpc_out=. --go-grpc_opt=paths=source_relative     ./metadata/proto/metadata.proto
+	python -m grpc_tools.protoc -I ./client/src --python_out=./client/src/ --grpc_python_out=./client/src/ ./client/src/featureform/proto/metadata.proto
+
+update_python: create_venv gen_grpc 			## Updates the python package locally
+	cd test_venv/bin && source activate
+	pip3 install pytest
+	pip3 install build
+	pip3 uninstall featureform  -y
+	-rm -r client/dist/*
+	python3 -m build ./client/
+	pip3 install client/dist/*.whl
+
+
+containers:						## Build Docker containers for Minikube
+	minikube image build -f ./api/Dockerfile . -t local/api-server:stable & \
+	minikube image build -f ./dashboard/Dockerfile . -t local/dashboard:stable & \
+	minikube image build -f ./coordinator/Dockerfile . -t local/coordinator:stable & \
+	minikube image build -f ./metadata/Dockerfile . -t local/metadata:stable & \
+	minikube image build -f ./metadata/dashboard/Dockerfile . -t local/metadata-dashboard:stable & \
+	minikube image build -f ./newserving/Dockerfile . -t local/serving:stable & \
+	wait; \
+	echo "Build Complete"
+
+start_minikube:	##Starts Minikube
+	minikube start
+
+reset_minikube:	##Resets Minikube
+	minikube delete
+	minikube start
+
+install_featureform: start_minikube containers		## Configures Featureform on Minikube
+	helm repo add jetstack https://charts.jetstack.io
+	helm repo update
+	helm install certmgr jetstack/cert-manager \
+        --set installCRDs=true \
+        --version v1.8.0 \
+        --namespace cert-manager \
+        --create-namespace
+	helm install featureform ./charts/featureform --set global.repo=local --set global.pullPolicy=Never --set global.version=stable
+	kubectl get secret featureform-ca-secret -o=custom-columns=':.data.tls\.crt'| base64 -d > tls.crt
+	export FEATUREFORM_HOST="localhost:443"
+    export FEATUREFORM_CERT="tls.crt"
+
+test_e2e: update_python					## Runs End-to-End tests on minikube
+	cd test_venv/bin && source activate
+	cd ../../
+	pip3 install requests
+	-helm install quickstart ./charts/quickstart
+	kubectl wait --for=condition=complete job/featureform-quickstart-loader --timeout=360s
+	kubectl wait --for=condition=READY=true pod -l app.kubernetes.io/name=ingress-nginx --timeout=360s
+	kubectl wait --for=condition=READY=true pod -l app.kubernetes.io/name=etcd --timeout=360s
+	kubectl wait --for=condition=READY=true pod -l chart=featureform --timeout=360s
+
+	-kubectl port-forward svc/featureform-ingress-nginx-controller 8000:443 7000:80 &
+	-kubectl port-forward svc/featureform-etcd 2379:2379 &
+
+	sleep 2
```

Comment:
is this always enough time? Is this waiting to make sure the port is forwarded?

Reply:
Updated